modules/fedimint-ln-client/src/lib.rs

```diff
@@ -910,6 +913,8 @@ impl LightningClientModule {
     /// See also [`Self::update_gateway_cache_continuously`].
     pub async fn update_gateway_cache(&self) -> anyhow::Result<()> {
         let gateways = self.module_api.fetch_gateways().await?;
+        // held to prevent transaction conflict
+        let _g = self.update_gateway_cache_lock.lock().await;
```

Comment:
Can't we do a try_lock before and if that failed return immediately after actually acquiring the lock and instantly releasing it since some else already updated the cache?

Reply:
baited me to properly implementing merging :smile:

Reply:
If the other thread failed, that would tell the caller things were updated while they weren't.

Reply:
now it is implemented in new struct, it is much better now